testplan/testing/base.py

```diff
@@ -441,34 +445,110 @@ def add_pre_resource_steps(self) -> None:
         self._add_step(self._build_environment)
         self._add_step(self._set_dependencies)
 
+    def _dry_run_start_resource(self) -> TestCaseReport:
+        suite_report = self.result.report[ResourceHooks.environment_start]
+        case_report = TestCaseReport(
+            name=ResourceHooks.starting,
+            category=ReportCategories.SYNTHESIZED,
+        )
+        suite_report.append(case_report)
+        return case_report
+
+    def _start_resource(self) -> TestCaseReport:
+        case_report = self._dry_run_start_resource()
+        case_result = self.cfg.result(
+            stdout_style=self.stdout_style, _scratch=self.scratch
+        )
+        self.resources.start()
+
+        for uid, driver in self.resources.items():
+            case_result.log(
+                f"{driver.__class__.__name__}[{uid}] Status: {driver.status.tag}"
+            )
+        case_report.extend(case_result.serialized_entries)
+        case_report.attachments.extend(case_result.attachments)
+        case_report.pass_if_empty()
+        return case_report
+
+    def _dry_run_stop_resource(self) -> TestCaseReport:
+        suite_report = self.result.report[ResourceHooks.environment_stop]
+        case_report = TestCaseReport(
+            name=ResourceHooks.stopping,
+            category=ReportCategories.SYNTHESIZED,
+        )
+        suite_report.append(case_report)
+        return case_report
+
+    def _stop_resource(self, is_reversed=False) -> TestCaseReport:
+        case_report = self._dry_run_stop_resource()
+        case_result = self.cfg.result(
+            stdout_style=self.stdout_style, _scratch=self.scratch
+        )
+        self.resources.stop(is_reversed)
+
+        for uid, driver in self.resources.items():
+            case_result.log(
+                f"{driver.__class__.__name__}[{uid}] Status: {driver.status.tag}"
+            )
+
+        case_report.extend(case_result.serialized_entries)
+        case_report.attachments.extend(case_result.attachments)
+        case_report.pass_if_empty()
+        return case_report
+
+    def _add_resource_report(self, name):
+        suite_report = TestGroupReport(
+            name=name, category=ReportCategories.SYNTHESIZED
+        )
+
+        if self.result.report.has_uid(name):
+            self.result.report[name] = suite_report
+        else:
+            self.result.report.append(suite_report)
+
+    def _finish_resource_report(self, name):
+        self.result.report[name].runtime_status = RuntimeStatus.FINISHED
+
     def add_start_resource_steps(self) -> None:
+        self._add_step(
+            self._add_resource_report, name=ResourceHooks.environment_start
```

Comment:
suggest that we add environment_start/stop testsuite in _run_resource_hook when not present --> then we have less steps and helps readability